admin/python/lsst/qserv/admin/nodePool.py

```diff
@@ -0,0 +1,160 @@
+# LSST Data Management System
+# Copyright 2015 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+
+"""
+Module defining nodePool class and related methods.
+
+NodePool is a class which allow to launch parallel ssh commands to
+a set of NodeAdmin instances. Each NodeAdmin instance can be a Qserv
+master or worker node.
+
+It has been used at CC-IN2P3 in spring 2015 to install and manage multiple
+Qserv development versions for a 50 nodes cluster.
+
+@author  Fabrice Jammes, IN2P3
+"""
+
+#--------------------------------
+#  Imports of standard modules --
+#--------------------------------
+import logging
+import os
+from subprocess import Popen, list2cmdline
+import time
+
+#-----------------------------
+# Imports for other modules --
+#-----------------------------
+
+#----------------------------------
+# Local non-exported definitions --
+#----------------------------------
+_LOG = logging.getLogger(__name__)
+
+def cpu_count():
+    ''' Returns the number of CPUs in the system
+    '''
+    try:
+        return os.sysconf('SC_NPROCESSORS_ONLN')
```

Comment:
` multiprocessing.cpu_count()` is probably more portable (in case IN2P3 switches to Windows:)